karb/parser.js

```diff
@@ -27,16 +27,22 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     }
     if (param.id) {
       result.unitid = param.id;
-      result.rtype  = 'ARTICLE';
     }
     if (param.format && param.format == 'pdf') {
       result.unitid = param.ids;
-      result.rtype  = 'ARTICLE';
       result.mime   = 'PDF';
     }
+
     if (param.journal) {
       result.unitid = param.journal;
     }
+
+    if ((param.id && param.id.includes('Ch')) || (param.ids && param.ids.includes('Ch'))) {
```

Comment:
Do you have an example for this case, to rule out any concerns about side effects and find a solution perhaps more suitable.